### DISCUSSION TITLE
Validating different message types in one file

### DISCUSSION THREAD

**Discussion by Kaiawaia** [xlate/staedi#401] [Q&A] [answered]
Hi,

I have to process edifact input files that contain messages with two different message types (VWAT and VWDT):

```
UNH+123456789000001+VWAT:01:0:0+111111111'
...VWAT segments...
UNT+3+123456789000001'
```
and
```
UNH+123456789000001+VWDT:01:0:0+111111111'
...VWDT segments...
UNT+3+123456789000001'
```
Now I want to set the transactionSchema in the START_TRANSACTION event. But I need two different schema files. One for VWAT and another one for VWDT. I've tried to get the message type in then START_TRANSACTION event, but without success. I need to set the transactionSchema depending on the message type. How can I accomplish this task?

Thanks and regards
Kai

**Comment by MikeEdgar** [comment, marked as answer]
@Kaiawaia , you can set the transaction schema anytime during the `UNH` segment. So, your code can retrieve and set the schema when it encounters the element with the transaction code.

**Reply by Kaiawaia** [reply]
Many thanks. That solved my problem.
